opt/scopeserver/dataserver/utils/loom.py

```diff
@@ -416,6 +417,32 @@ def add_user_clustering(self, request) -> Tuple[bool, str]:
             logger.debug("Failure")
             return (False, "Updated clusterings do not exist in file. Write error.")
 
+    def get_cluster_overlaps(self, cell_idxs):
+        metadata = self.get_meta_data()
+        cluster_overlap_data = []
+        if not self.has_md_clusterings:
+            return []
+        for clustering_meta in metadata["clusterings"]:
+            clustering_name = clustering_meta["name"]
+            clustering_id = clustering_meta["id"]
+            clustering = self.get_clustering_by_id(clustering_id)
+            counts = Counter(clustering[cell_idxs])
+            all_counts = Counter(clustering)
+            for cluster_id in counts.keys():
+                cluster_name = self.get_meta_data_cluster_by_clustering_id_and_cluster_id(
```

Comment:
This requires a check to pass on a `cluster_id` of `-1`. `-1` is added by custom clustering when a cell is not in any of the clusters, but is not present in the metdata. Given this may come up more in the future, the exception may be better handled in the `get_meta_data_cluster_by_clustering_id_and_cluster_id` function